src/hotspot/share/prims/jvmtiThreadState.cpp

```diff
@@ -404,8 +422,17 @@ JvmtiVTMSTransitionDisabler::VTMS_transition_enable_for_all() {
 void
 JvmtiVTMSTransitionDisabler::start_VTMS_transition(jthread vthread, bool is_mount) {
   JavaThread* thread = JavaThread::current();
+  oop vt = JNIHandles::resolve_external_guard(vthread);
+
+  if (!sync_protocol_enabled()) {
```

Comment:
Isn't the check racy? So we could see that there is no disabler, but before setting anything a disabler can run and succeed. Then we would never notice it and will just return after setting these fields.
I think you need to check this after setting _VTMS_transition_count and the vthread transition bit.

Reply:
Thank you for looking at this, Patricio!
Nice catch - fixed now.